pulse2percept/stimuli/base.py

```diff
@@ -0,0 +1,465 @@
+
+        """
+        if not self._need_interp():
+            # This includes the special case of a single time point (where
+            # `interp1d` would not work):
+            return Stimulus(self.data[:, 0], electrodes=self.electrodes,
+                            time=None, compress=self.compressed)
+
+        # Should work with scalars and lists:
+        time = np.array([time]).flatten()
+        data = np.array([[self._interp[e](t) for t in time]
+                         for e, _ in enumerate(self.electrodes)])
+        return Stimulus(data, electrodes=self.electrodes, time=time,
+                        compress=self.compressed)
+
+    def __eq__(self, other):
+        """Returns True if two Stimulus objects are identical
+
+        Two Stimulus objects are considered identical if they have the same
+        electrode names, time steps, and data points.
+
+        Parameters
+        ----------
+        other : any
+            Another object or variable to which the current object should be
+            compared.
+
+        Examples
+        --------
+        >>> from pulse2percept.stimuli import Stimulus
+        >>> stim1 = Stimulus(np.ones(3))
+        >>> stim2 = Stimulus(np.zeros(5))
+        >>> stim1 == stim2
+        False
+
+        Compare a Stimulus with something else entirely:
+
+        >>> from pulse2percept.stimuli import Stimulus
+        >>> stim1 = Stimulus(np.ones(3))
+        >>> stim1 == 1
+        False
+
+        """
+        if not isinstance(other, Stimulus):
+            return False
+        if self.time is None:
+            if other.time is not None:
+                return False
+        else:
+            if other.time is None:
+                return False
+            if len(self.time) != len(other.time):
+                return False
+            if not np.allclose(self.time, other.time):
+                return False
+        if len(self.electrodes) != len(other.electrodes):
+            return False
+        if not np.all(self.electrodes == other.electrodes):
+            return False
+        if self.shape != other.shape:
+            return False
+        if not np.allclose(self.data, other.data):
+            return False
+        return True
+
+    def __ne__(self, other):
+        """Returns True if two Stimulus objects are different
+
+        Two Stimulus objects are considered different if they store different
+        electrode names, time steps, or data points.
+
+        Parameters
+        ----------
+        other : any
+            Another object or variable to which the current object should be
+            compared.
+
+        Examples
+        --------
+        Compare two Stimulus objects:
+
+        >>> from pulse2percept.stimuli import Stimulus
+        >>> stim1 = Stimulus(np.ones(3))
+        >>> stim2 = Stimulus(np.zeros(5))
+        >>> stim1 != stim2
+        True
+
+        """
+        return not self.__eq__(other)
```

Comment:
Would be neat to implement the `__get_item__` that takes a time point and returns the data at that time point, including interpolating as needed. But maybe on a separate PR.

Reply:
I added this feature request in issue #121